src/Discord.Net.Commands/CommandService.cs

```diff
@@ -287,7 +287,17 @@ public Task<IResult> ExecuteAsync(ICommandContext context, int argPos, IServiceP
 
             foreach (var match in commands)
             {
-                preconditionResults[match] = await match.Command.CheckPreconditionsAsync(context, services).ConfigureAwait(false);
+                try
```

Comment:
I feel like the lib shouldn't be doing this. Let the precondition itself catch-and-pass any expected exceptions to the result object.

Reply:
I could add a `CommandService` option that defaults to `true` on whether or not preconditions should throw to match with `ThrowOnError`. That way preconditions would by default throw, but a command service option would allow the lib to return a exception precondition, which makes it easier to have a centralized place to handle all exceptions.